h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java

```diff
@@ -71,6 +73,28 @@ public void init(boolean expensive) {
     super.init(expensive);
 
     checkFoldColumnPresent(_parms._metalearner_fold_column, train(), valid(), _parms.blending());
+    validateAndExpandBaseModels();
+  }
+
+  /**
+   * Validates base models and if grid is provided instead of a model it gets expanded.
+   */
+  private void validateAndExpandBaseModels() {
+    List<Key> base_models = new ArrayList<Key>();
+    for (Key base_model: _parms._base_models) {
+      Object retrieved_object = DKV.getGet(base_model);
+      if (retrieved_object instanceof Model) {
+        base_models.add(base_model);
+      } else if (retrieved_object instanceof Grid) {
+        Grid grid = (Grid) retrieved_object;
+        Collections.addAll(base_models, grid.getModelKeys());
+      } else if (retrieved_object == null) {
+        continue;
```

Comment:
please throw exception here as well: principle of less surprise.
If user specifies an id that doesn't exist, we should tell him immediately.